database_operations.py

```diff
@@ -104,4 +108,50 @@ def delete_notified_reminders():
         logger.error(f"Se produjo un error al eliminar recordatorios notificados: {e}", exc_info=True)
     finally:
         session.close()
-        
\ No newline at end of file
+
+class AnalisisButton(discord.ui.Button):
+    def __init__(self, message_id, channel_id):
+        super().__init__(label="Análisis", style=discord.ButtonStyle.grey)
+        self.message_id = message_id
+        self.target_channel_id = channel_id
+
+
+    async def callback(self, interaction: discord.Interaction):
+        # Intenta obtener el thread_id asociado con el message_id.
+
+        thread_id = self.message_id # The message_id is == thread_id
+
+        if thread_id:
+            # Si hay un thread_id asociado, realiza la lógica para copiar mensajes al nuevo hilo
+            original_thread = interaction.client.get_channel(int(thread_id))
+            target_channel = interaction.client.get_channel(self.target_channel_id)
+
+            if original_thread:
+                new_thread = await target_channel.create_thread(name= f"Análisis de Mensaje {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}" )
+                await copy_messages_to_new_thread(interaction.client, thread_id, new_thread)
+                await interaction.response.send_message(
+                    f"Los mensajes del hilo {original_thread.name} se han copiado al hilo {new_thread.name} en el canal {target_channel.name}",
+                    ephemeral=True
+                    )
+            else:
+                await interaction.response.send_message("El hilo original no está disponible.", ephemeral=True)
+        else:
+            await interaction.response.send_message(f"No hay un hilo específico asociado para análisis. {self.message_id} ", ephemeral=True)
+
+async def copy_messages_to_new_thread(bot, original_thread_id, new_thread: discord.Thread):
+    try:
+        original_thread = bot.get_channel(original_thread_id)
+        if original_thread:
+            # Verifica si original_thread es realmente un Thread y no un Channel regular
+            if isinstance(original_thread, discord.Thread):
+
+                async for message in original_thread.history(limit=123, oldest_first=True):
+                    if message.content:
```

Comment:
1. The older_first=True parameter will instruct the history function to retrieve the oldest message first, ensuring it becomes the first message sent to the new thread.

2. The if message.content statement is used to verify whether the message contains content. This also helps prevent the HTTPError 'cannot send blank message' from occurring.